lib/resourcemerge/core.go

```diff
@@ -56,6 +56,16 @@ func ensurePodSpec(modified *bool, existing *corev1.PodSpec, required corev1.Pod
 		}
 		ensureContainer(modified, existingCurr, required)
 	}
+	var requiredContainers []corev1.Container
+	for _, existing := range existing.Containers {
+		for _, required := range required.Containers {
+			if existing.Name == required.Name {
+				requiredContainers = append(requiredContainers, existing)
+				continue
+			}
+		}
+	}
+	existing.Containers = requiredContainers
```

Comment:
The entire work of ensure containers done https://github.com/openshift/cluster-version-operator/pull/178/files#diff-f7675c7b39553e431cdc6c3ba8f5c482R44 will be unused due to this.

Also this will keep forcing update of containers because required never has any server side defaults.

What you really want is remove any containers that are not part of required.

Reply:
Ahh, nvm it sets the existing correctly. I missed the inner loop

Reply:
> The entire work of ensure containers done https://github.com/openshift/cluster-version-operator/pull/178/files#diff-f7675c7b39553e431cdc6c3ba8f5c482R44 will be unused due to this.

Thanks for pointing out. Will move this logic before ensureContainers to save cycles.

> Also this will keep forcing update of containers because required never has any server side defaults.

I am assuming you meant `server side default init containers`. If correct, we are removing init containers based on required. Is this common to have server side default regular containers?

Reply:
> If correct, we are not removing init containers based on required.

Should we have a single `ensureContainers` that we run on both `InitContainers` and `Containers`?